nevergrad/optimization/recaster.py

```diff
@@ -206,3 +207,77 @@ class SequentialRecastOptimizer(RecastOptimizer):
     # pylint: disable=abstract-method
 
     no_parallelization = True
+
+
+class BatchRecastOptimizer(RecastOptimizer):
+    """Recast optimizer where points to evaluate are provided in batches
+    and stored by the optimizer to be asked and told on. The fake_callable
+    is only brought into action every 'batch size' number of asks and tells
+    instead of every ask and tell. This opens up the optimizer to
+    parallelism.
+
+    Note
+    ----
+    These implementations are not necessarily robust. You have to complete
+    a batch before you start a new one so parallelism is only possible
+    within batches i.e. if a batch size is 100 and you have done 100 asks,
+    you must do 100 tells before you ask again but you could do those 100
+    asks and tells in parallel.
+
+    """
+
+    # pylint: disable=abstract-method
+
+    def __init__(
+        self, parametrization: IntOrParameter, budget: tp.Optional[int] = None, num_workers: int = 1
+    ) -> None:
+        super().__init__(parametrization, budget, num_workers=num_workers)
+        self._current_batch: tp.List[p.Parameter] = []
+        self._batch_losses: tp.List[tp.Loss] = []
+        self._tell_counter = 0
+        self.batch_size = 0
+        self.indices: tp.Dict[p.Parameter, int] = {}
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        """Reads messages from the thread in which the underlying optimization function is running
+        New messages are sent as "ask".
+        """
+        # get a datapoint that is a random point in parameter space
+        if self._messaging_thread is None:
+            self._messaging_thread = MessagingThread(self.get_optimization_function())
+        # wait for a message
+        if not self._messaging_thread.is_alive():  # In case the algorithm stops before the budget is elapsed.
+            warnings.warn(
+                "Underlying optimizer has already converged, returning random points",
+                base.errors.FinishedUnderlyingOptimizerWarning,
+            )
+            self._check_error()
+            data = self._rng.normal(0, 1, self.dimension)
+            return self.parametrization.spawn_child().set_standardized_data(data)
+        # if no more points left in batch, wait for new batch from fake callable
+        if not self._current_batch:
+            points = self._messaging_thread.messages_ask.get()
+            self.batch_size = len(points)
+            self._current_batch = [self.parametrization.spawn_child(point) for point in points]
```

Comment:
```suggestion
            self._current_batch = [self.parametrization.spawn_child().set_standardized_data(point) for point in points]
```
The value parameter in spawn_child has a different meaning (aka the actual value for the function, which could be a string, a dict whatever, while optimizers work in a standardized space)